[Jagdeep Singh Hundal] iP

jagdeepsh opened the pull request:
# Jag's Chat Bot

This bot features:
- Fast user interface
- Coded in `Java`
- Simplicity in commands

This bot is meant for:
1. Anyone that would like to keep track of their tasks
2. Anyone that loves to organise their todo's
3. Perhaps someone that has a lot of tasks to do

Here is my `main` method's contractor called `jag` for a quick reference to how it works:
```java
public Jag(String path) {
    ui = new Ui();
    storage = new Storage(path);
    try {
        tasks = storage.load();
    } catch (FileNotFoundException e) {
        ui.showLoadingError();
        tasks = new TaskList();
    }
}
```

Here are some of the tasks that the `Java` programme is able and unable to do 👍:
- [x] Create todo's type tasks
- [x] Create deadlines type tasks
- [x] Create event type tasks
- [x] mark tasks
- [x] unmark tasks
- [x] delete tasks
- [x] list tasks
- [x] find tasks
- [ ] Anything else not mentioned above, is excluded from the chat bot's capabilities

A famous quote I would like to share for managing tasks are as follows:
> "Be like a postage stamp, stick to one thing until you get there" - Josh Billings


**Check out my link to my repo:**
[Here is the link to my repo](https://github.com/JagdeepSinghNUS/ip)

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

jagdeepsh committed:
Update Jag.java

jagdeepsh committed:
Update Jag.java Rename, Greet, Exit

jagdeepsh committed:
Update Jag.java Echo

jagdeepsh committed:
Update Jag.java Add, List

jagdeepsh committed:
Update Jag.java Mark as Done

jagdeepsh committed:
Update Jag.java ToDos, Events, Deadlines

jagdeepsh committed:
A-TextUiTesting

jagdeepsh committed:
Add AException.java

jagdeepsh committed:
Update Jag.java Delete

jagdeepsh committed:
Add Commands.java

jagdeepsh committed:
Save

jagdeepsh committed:
Merge branch 'branch-Level-7'

jagdeepsh committed:
Dates and Times

jagdeepsh committed:
Merge branch 'branch-Level-8'

jagdeepsh committed:
A-MoreOOP

jagdeepsh committed:
Merge branch 'branch-A-MoreOOP'

jagdeepsh committed:
Add packages

jagdeepsh committed:
Merge branch 'branch-A-Packages'

jagdeepsh committed:
Merge branch 'add-gradle-support'

jagdeepsh committed:
Implement JUnit

jagdeepsh committed:
Add JUnit

jagdeepsh committed:
Merge branch 'branch-A-JUnit'

jagdeepsh committed:
Update deleteCommand

jagdeepsh committed:
Add JAR File

jagdeepsh committed:
Merge branch 'branch-A-Jar'

jagdeepsh committed:
Add JavaDoc

jagdeepsh committed:
Merge branch 'branch-A-JavaDoc'

jagdeepsh committed:
Add Coding Standard

jagdeepsh committed:
Merge branch 'branch-A-CodingStandard'

jagdeepsh committed:
Update JavaDoc

jagdeepsh committed:
Add find command

jagdeepsh committed:
Conflict resolved

jagdeepsh committed:
Update JavaDoc

jagdeepsh committed:
Add CheckStyle

jagdeepsh committed:
Merge branch 'branch-A-CheckStyle'

jagdeepsh committed:
Add GUI feature

rahula1008 commented on src/main/java/jag/AddCommand.java:
```diff
@@ -0,0 +1,110 @@
+     *              the while loop
+     */
+    @Override
+    public Boolean isExit() {
```
I like that you aptly named your boolean methods according to the coding standard.

rahula1008 commented on src/main/java/jag/Event.java:
```diff
@@ -0,0 +1,33 @@
+package jag;
```
Could it be a good idea to keep the events and tasks classes in separate packages?

rahula1008 commented on src/main/java/jag/Commands.java:
```diff
@@ -0,0 +1,9 @@
+ * to find the right command entered by the user in a Switch statement
+ */
+public enum Commands {
+    LIST, BYE, MARK, UNMARK, TODO, EVENT, DEADLINE, DELETE, FIND
```
Small nit, but is it possible that these enums should be in camel case? It mentions that Class/enum names must be nouns and written in PascalCase.

rahula1008 commented on src/main/java/jag/FindCommand.java:
```diff
@@ -0,0 +1,47 @@
+     * Call upon the Ui instance to get the description that is
+     * to be found,
+     *
+     * @param tasks Tasklist instance to be used to find any task that
```
Small nit here, but could it be more readable if you wrote your descriptions of the params more concisely?

rahula1008 commented on src/main/java/jag/Parser.java:
```diff
@@ -0,0 +1,75 @@
+     * @throws AExceptions Instance thrown in the event user input does not include
+     *              any of the right commands available
+     */
+    public static Command parse(String fullCommand) throws AExceptions {
```
I like that you use verbs for your methods.

rahula1008 commented on src/main/java/jag/Parser.java:
```diff
@@ -0,0 +1,75 @@
+        // Breaking down command input
+        String[] splitWords = fullCommand.split(" ");
+        String com = splitWords[0].toUpperCase();
+        Commands command;
```
Perhaps this command variable could be replaced with commands, to show how it represents multiple enums.
```suggestion
        Commands commands;
```

rahula1008 commented on src/main/java/jag/TaskList.java:
```diff
@@ -0,0 +1,44 @@
+     * an instance of ArrayList, which will be used to store any given list
+     * of tasks
+     */
+    public TaskList() {
```
I like that you use nouns for your class names and pascal case as well.

rahula1008 commented on src/main/java/jag/Ui.java:
```diff
@@ -0,0 +1,288 @@
+     * @return an Integer representing the index of the task to be marked
+     *          or unmarked in the stored TaskList
+     */
+    public int getMark() {
```
Nit here, but could this be renamed to getMarkIndex so to more explain the named entity to the reader in more detail?

rahula1008 commented on src/main/java/jag/Ui.java:
```diff
@@ -0,0 +1,288 @@
+     * Displays the response for exiting the chatbot
+     * and to close the instance of the scanner to end the application
+     */
+    public void exitResponse() {
```
Another nit here, could this be renamed to something like displayExitResponse to more clearly resemble the use of a verb and explain more clearly?

rahula1008 reviewed (commented):
Overall, I like your code and it reads very well. I think there are some moments where the Javadocs could be made more concise, and it could be possible to organise your code into more sub-packages, if that makes sense. But overall, I think you adhere to the coding standards and use good names for your boolean methods/variables.

jagdeepsh committed:
Update GUI feature

jagdeepsh committed:
Merge branch 'branch-Level-10'

jagdeepsh committed:
Add Java Varargs

jagdeepsh committed:
Merge branch 'branch-A-Varargs'

chuajunyu commented on src/test/java/jag/ParserTest.java:
```diff
@@ -0,0 +1,80 @@
+    }
+
+    @Test
+    public void parse_todods_test() throws AExceptions {
```
nit: perhaps using this three part naming convention format mentioned in the java conventions for naming your test case methods could make them more descriptive?

3 part format: `featureUnderTest_testScenario_expectedBehavior()`

chuajunyu commented on src/main/java/jag/UiCLI.java:
```diff
@@ -0,0 +1,337 @@
+ * Purpose is to read messages / commands, respond appropriately,
+ * and return values to be used for other Command instances e.g. "Event"
+ */
+public class UiCLI extends Ui {
```
Since abbrieviations and acronyms should not be uppercase when used as part of a name, perhaps this could be renamed to something like UiCli?

chuajunyu commented on src/main/java/jag/DeleteCommand.java:
```diff
@@ -0,0 +1,54 @@
+     *           of the task to be deleted in the TaskList instance
+     * @param storage To update the output file storage after task
+     *                has been deleted
+     */
```
Nit: Note that there should be punctuation (period) after each description.

chuajunyu commented on src/main/java/jag/Command.java:
```diff
@@ -0,0 +1,10 @@
+package jag;
```
Would it be a better idea to organise all of the Command classes into a commands package instead?

chuajunyu commented on src/main/java/jag/DeleteCommand.java:
```diff
@@ -0,0 +1,54 @@
+     *                has been deleted
+     */
+    @Override
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
```
I like how you used plural form on TaskList as it represents a collection of Tasks!

chuajunyu commented on src/main/java/jag/Storage.java:
```diff
@@ -0,0 +1,128 @@
+    public TaskList load() throws FileNotFoundException {
+        File f = new File(this.filePath);
+        char taskType;
+        boolean isDone;
```
I like how you named your booleans to sound like booleans!

chuajunyu commented on src/main/java/jag/Parser.java:
```diff
@@ -0,0 +1,75 @@
+        try {
+            command = Commands.valueOf(com);
+
+            switch (command) {
```
I like how you used the right indentation for the switch cases!

chuajunyu reviewed (commented):
Overall, I think you have adhered really well to the coding standards! I like your descriptive and apt naming conventions. Some small improvements could be to organise your classes into more packages for easier maintainence of your codebase in the long run, as well as add punctuation to your header comments. Good job!

jagdeepsh committed:
Update Jag

jagdeepsh committed:
Add Assertions to Jag Parser and AddCommand

To reduce potential errors as code currently has no Assertions

Add assertions to check for valid string lengths and instance types

Assertions added to debug errors from assumptions

No other changes are included in this commit.

jagdeepsh committed:
Add assertions in FindCommand

jagdeepsh committed:
Fix Code Quality

Version of code had repeated code and comments explaining How

Refactored code by using showLine() to print dashed lines

Add readNextCommand() method in UiCLI to apply DRY and abstraction

Cleaned up JavaDocs comment to exclude How

To keep comments to the point and short and to not repeat lines of code

jagdeepsh committed:
Add Java Streams in Find

jagdeepsh committed:
Merge pull request #1 from JagdeepSinghNUS/branch-A-Assertions

branch-A-Assertions

jagdeepsh committed:
Merge branch 'master' into branch-A-CodeQuality

jagdeepsh committed:
Merge branch 'master' into branch-A-Streams

jagdeepsh committed:
Merge pull request #2 from JagdeepSinghNUS/branch-A-CodeQuality

branch-A-CodeQuality

jagdeepsh committed:
Merge branch 'master' into branch-A-Streams

jagdeepsh committed:
Merge pull request #3 from JagdeepSinghNUS/branch-A-Streams

branch-A-Streams

jagdeepsh committed:
Add Workflows yml file

jagdeepsh committed:
Merge branch 'branch-A-CI'

jagdeepsh committed:
Add C-Update

jagdeepsh committed:
Merge pull request #4 from JagdeepSinghNUS/branch-C-Update

Add C-Update

jagdeepsh committed:
Add JavaDoc in UpdateCommand

jagdeepsh committed:
Merge pull request #5 from JagdeepSinghNUS/branch-C-Update

Add JavaDoc in UpdateCommand

jagdeepsh committed:
Update README file

jagdeepsh committed:
Merge branch 'branch-A-UserGuide'

jagdeepsh committed:
Update README

jagdeepsh committed:
Update Storage

jagdeepsh committed:
Merge branch 'branch-A-Release'
